src/web/mjs/connectors/Ridibooks.mjs

```diff
@@ -0,0 +1,247 @@
+        // as of now (11 November 2023), Ridibooks
+        // still uses majoritarily the legacy API
+        // with graphQL used only for a few use cases.
+        // Webtoons not being one of them;
+        let mangaListGql = await this._getMangasGql();
+        mangaList.push(...mangaListGql);
+
+        return mangaList;
+    }   
+
+    /**
+     * Method to get all chapters for a webtoon.
+     * @param {IManga} manga
+     * @returns {IChapter[]}
+     * @override
+     */
+    async _getChapters(manga) {
+        let uri = new URL(`/books/${manga.id}`, this.url);
+
+        let request = new Request(uri, this.requestOptions);
+
+        return Engine.Request.fetchUI(request, 'seriesBookListJson')
+            .then(data => data.map(item => ({
+                id: item.id,
+                title: item.title
+            })));
+    }
+
+    /**
+     * Method to get all pages of a chapter.
+     * @param {Chapter} chapter
+     * @returns {Promise<string[]>}
+     * @override
+     */
+    async _getPages(chapter) {
+        let uri = new URL(
+            `/generate/${chapter.id}`,
+            this.pagesApi
+        );
+
+        let request = new Request(uri, this.requestOptions);
+
+        return this.fetchJSON(request, 3)
+            .then(res => {
+                if (!res.success) {
+                    res.login_required && 
+                        alert("Login required.");
+                    return [];
+                }
+
+                return res.pages.map(page => page.src);
+            });
+    }
+
+    /**
+     * Get the manga from the provided resource.
+     * @param {URL|string} uri
+     * @returns {Promise<Manga>}
+     * @override
+     */
+    async _getMangaFromURI(uri) {
+        if (!this.isBookPageURL(uri)) {
+            alert("Not a valid URL.");
+            return;
+        }
+
+        // first check if it already exists in cache
+        let id = this.getIdFromUri(uri);
+        let manga = await this.findMangaById(id);
+        if (manga) return manga;
+
+        // else, try new request
+        let request = new Request(uri, this.requestOptions);
+
+        return Engine.Request.fetchUI(request, "bookDetail")
+            .then(data => new Manga(
+                this,
+                data.series_id,
+                data.series_title
+            ));
+    }
+
+    // -----------------------------
+
+    /** 
+     * Helper function that checks if a given URI is a valid book page URL for the Ridibooks website.
+     * @param {string|URL} uri - The URI from which to extract the ID.
+     * @returns {boolean}
+     */
+    isBookPageURL(uri) {
+        return this.bookUrlPattern.test(String(uri));
+    }
+
+    /**
+     * Extracts the ID from a URI.
+     * 
+     * @param {string|URL} uri - The URI from which to extract the ID.
+     * @returns {string|undefined} The extracted ID, or undefined if no match is found.
+     */
+    getIdFromUri(uri) {
```

Comment:
I *really* disagree with this statement. Just because a function is used only once doesn't mean it's unnecessary. It's about separation of concerns and clean code. Anyway, the part of the code where I used it will be removed so I'll remove it too, but not because it's "unnecessary".

Reply:
Called once + 1 liners functions is a pain.
as a maintainer, it adds a burden when there is an issue on the connector and you have to dig through multiple smalls functions.

Reply:
I'm not trying to be combative, but I wholeheartedly disagree. I'm pretty sure your editor of choice has a way to go to a function's definition and to return to the previous position. It's a non-issue. But anyway, like I said, I'm removing this.

Reply:
I see your point , but code is clean enough. Too much decomposition is painful and we are accustomed to inline stuff like that. And using a function for something called once is overkill. We wouldnt mind if it was even only call twicev  ofc.